Send websocket notification on collaborator accept and remove

TimothyGillespie opened the pull request:
On the request of @NSGolova. Seems to make a lot of sense to me.

Top-Cat commented on src/jvmMain/kotlin/io/beatmaps/api/collaboration.kt:
```diff
@@ -119,6 +120,8 @@ fun Route.collaborationRoute() {
                             it[accepted] = true
                         }
 
+                        call.pub("beatmaps", "maps.${map.id}.updated.collaborators.added", null, map.id.value)
```
Calling a publish from inside a transaction doesn't guarantee the transaction will be committed before the message is processed.

TimothyGillespie replied:
You are right! I now pulled it out of the transaction - I hope they way I did is fine. Otherwise, happy to hear suggestions!

Top-Cat reviewed (changes requested):
What's the thinking behind having both `collaborators.added` and `collaborators.removed`? As they're not processed differently it's currently just adding complication.

TimothyGillespie commented:
I did removed the `.added` and `.removed` respectively.

The reason why I did that is because it seemed "natural" to namespace it like that and if we were to refactor/change or utilize these somehow it'd be easier. I do find it to be clearer that way as to what is included etc.

But realistically, this will never get used and I see your point. Might have some negative performance or resource impact as well.

Top-Cat commented on src/jvmMain/kotlin/io/beatmaps/api/collaboration.kt:
```diff
@@ -152,17 +154,20 @@ fun Route.collaborationRoute() {
                         id eq req.collaborationId and (collaboratorId eq sess.userId)
                     } > 0
+
+                    Pair(success, null)
```
But doesn't this mean that the message won't be sent? map == null
Given you can't get a map object here it might make more sense for the second result to also be a boolean and return `map != null` above instead of `map`

You'll also want to use `req.mapId` below instead of using the map object.

TimothyGillespie replied:
Exactly! This is pretty much intended.

This is the case where the collaboration is rejected, hence no change. I find it bit awkward too though, best I could think of define the returned second parameter to be null if not successful accepted.

I did try to use `req.mapId`, but I quickly realized it doesn't exist, because the collaboration request object is denied or accepted and only that ID is passed.
